whois-query/src/main/java/net/ripe/db/whois/query/query/Query.java

```diff
@@ -136,12 +139,15 @@ public static Query parse(final String args, final UserSession userSession, fina
     }
 
 
-    public static Query parse(final String args, final UserSession userSession, final List<String> passwords, final boolean trusted, final List<X509CertificateWrapper> certificates, final OAuthSession oAuthSession) {
+    public static Query parse(final String args, final UserSession userSession, final List<String> passwords,
+                              final String override, final boolean trusted,
+                              final List<X509CertificateWrapper> certificates, final OAuthSession oAuthSession) {
         final Query query = parse(args, Origin.REST, trusted);
         query.userSession = userSession;
         query.passwords = passwords;
         query.certificates = certificates;
         query.oAuthSession = oAuthSession;
+        query.override = StringUtils.isEmpty(override) ? null : OverrideCredential.parse(override);
```

Comment:
I think what you should be checking is OverrideValues in OverrideCredential is present or not , it might very well happen that override value is wrongly formatted in that case we put optional null in OverrideValues. Just checking override == null is not enough ?

Reply:
I move that check into OverrideCredential